### ISSUE TITLE
Attaching additional graphical elements for displaying omics data

### ISSUE THREAD

**Issue by Midnighter** [zakandrewking/escher#273]
Hi,
For displaying omics data on reactions, for example, RNAseq, I would like to display different kind of elements instead of the textual GPR associations. Below is a handcrafted example prepared by @meono that colors each complex in multiple parts (different shades of color based on fold change) and multiple complexes if they exist.

I would like to enable something like that with a similar API to showing data on reactions/metabolites. Instead of the boxes other types of plots, like histograms, could also be interesting in future.

My questions is, would  a PR for this be welcome? If yes, do you have a suggestion where I should start on integrating this into Escher, what classes/functions to modify etc.?

![image](https://user-images.githubusercontent.com/135653/40791909-d2873bf0-64f8-11e8-9c0c-85ab708eba93.png)

**Comment by zakandrewking** [comment]
This would be awesome. I've been thinking about the need for a generic approach to embed arbitrary SVG next to reaction, metabolites, or genes. We might also want to create placeholders (e.g. an empty rectangle of a certain size) so that people who are building maps know to keep enough room for these annotations.

Can you say a little more about your needs? Is it OK to have it entirely in SVG? For the PR, are you thinking primarily about the API?

The best first step is for you to take a look at the source code (master branch) and let me know how much of it is self-explanatory and what high-level architectural questions you have. Then we can get into some details about how best to implement this API.

I'm glad your interested in contributing!

**Comment by Midnighter** [comment]
> Can you say a little more about your needs? Is it OK to have it entirely in SVG? For the PR, are you thinking primarily about the API?

To be honest as a first step I just want to get this capability in as quickly as possible (and by in I mean a local workable example). SVG works great for me. For the pull request, I guess you will have specific ideas what the API should look like.

**Comment by zakandrewking** [comment]
yeah, but i'm open to reviewing a quick-and-dirty prototype first

we settled on Preact Components to make the tooltips reusable, so that's
probably smart here too, unless the performance sucks
